modules/matrix.lobster

```diff
@@ -0,0 +1,42 @@
+import vec
+
+// A 2x2 matrix representated as a 4 length vector of x,y,z,w so that
+// |x y|
+// |z w|
+
+struct mat2x2 : xyzw_f
+
+    def operator*(o:mat2x2):
+        // This could be expressed with 4 dot product if `this.zw` existed (and 
+        // others)
+        return mat2x2 { x*o.x + y*o.z, x*o.y + y*o.w,
+                        z*o.x + w*o.z, z*o.y + w*o.w }  
+    
+    def operator*(o:xy_f):
+        // This would be nicer if `this.zw` existed.
+        return xy_f { dot(o, this.xy),
+                      dot(o, xy_f{z,w})}
+
+
+    def operator*(f:float):
```

Comment:
we could leave out this operator entirely?

Reply:
Sorry, the highlighting is odd. Which operator, the *float?

Reply:
yes this entire function, `def operator*(f:float)`

Reply:
Ok, removed. Those it does remove this `assert a * 5.0 == mat2x2 { 5.0, 10.0, 15.0, 20.0 }` from working.